### DISCUSSION TITLE
How to run via Docker

### DISCUSSION THREAD

**Discussion by xela-nosretap** [DIGITALCRIMINALS/OnlyFans#754] [Q&A]
I'm used to running things like Sonarr, Radarr etc via Docker, I haven't run anything like this in Docker.

I usually create a stack in Portainer when there are any new containers I want to run.

Do I need to download the code from GitHub and run the Docker command from in there, or is there a pre-made image I can just pull and use? I found this https://hub.docker.com/r/dockanon6573456/onlyfans, but I'm still no closer to getting it to run.

I understand there is no web interface for this and I will have to run the container in an interactive terminal to use it.

**Comment by ecchiecchi0** [comment]
There was just a big update that broke the docker build and I have submitted a PR with fixes but until it gets merged the docker method will fail to build/run.

All of the images on docker hub are out of date and may not work but since this is not something you probably want to leave constantly running it makes sense to just update and build the a new container every time your run it.

1. Go to the directory where you would like to build/run the container from.
2. Clone the repository and change to the OnlyFans directory that was created:
`$ git clone https://github.com/DIGITALCRIMINALS/OnlyFans`
`$ cd OnlyFans`

3. Update, build and run the container:
`$ git pull && sudo docker build -t only-fans . && sudo docker run -it --rm --name onlyfans -v ${PWD}/.settings:/usr/src/app/.settings -v ${PWD}/.profiles:/usr/src/app/.profiles -v ${PWD}/.sites:/usr/src/app/.sites only-fans`

During the first run the .settings, .profiles and .sites directories will be created and mapped to the container and it will ask you to modify your settings in the config files that get created to login to your account. When you want to run this again your just skip directly to the command in step 3 (run from the OnlyFans directory).

**Reply by ecchiecchi0** [reply]
Also, I am not that familiar with Portainer but it looks like it has the ability to automate building images from git repo sources so that might work for you too.
